src/sdk/pynni/nni/compression/torch/pruners.py

```diff
@@ -215,29 +215,31 @@ def calc_mask(self, layer, config):
             dictionary for storing masks
         """
 
-        weight = layer.module.weight.data
-        op_name = layer.name
-        op_type = layer.type
+        weight = wrapper.module.weight.data
+        op_name = wrapper.name
+        op_type = wrapper.type
         assert op_type == 'BatchNorm2d', 'SlimPruner only supports 2d batch normalization layer pruning'
-        if op_name in self.mask_calculated_ops:
-            assert op_name in self.mask_dict
-            return self.mask_dict.get(op_name)
+        # if op_name in self.mask_calculated_ops:
+        #     assert op_name in self.mask_dict
+        #     return self.mask_dict.get(op_name)
         base_mask = torch.ones(weight.size()).type_as(weight).detach()
-        mask = {'weight': base_mask.detach(), 'bias': base_mask.clone().detach()}
+        mask_weight = base_mask.detach()
+        mask_bias = base_mask.clone().detach()
         try:
             filters = weight.size(0)
-            num_prune = int(filters * config.get('sparsity'))
+            num_prune = int(filters * wrapper.config.get('sparsity'))
             if filters < 2 or num_prune < 1:
-                return mask
+                wrapper.weight_mask = mask_weight
```

Comment:
I think it's not a good idea to make wrapper mutable here. It's very hard to learn for users.

Reply:
In users' perspective, this function is simply calculating the mask of weight and bias, which has nothing to do with "wrapper".

Reply:
removed, return mask instead